File: week_2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");
    int count_letters = 0;
    int count_words = 1;
    int count_Sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            count_letters++;
        }
        if (text[i] == ' ')
        {
            count_words++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count_Sentences++;
        }
    }
    double L = (double)count_letters * 100 / (double)count_words;
    double S = (double)count_Sentences * 100 / (double)count_words;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    if (index - (int)index >= 0.5)
    {
        index++;
        printf("Grade %i\n", (int)index);
    }
    else if (index < 1)
    {
        index = 1;
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int)index);
    }

    return 0;
}

File: week_6_Python/readability/readability.py
# Get the text from user
print("Text: ", end='')
text = input()

# Initialize counters
count_letters = 0
count_words = 1
count_Sentences = 0

# Counting words, sentences, letters
for i in range(len(text)):
    if ((text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z')):
        count_letters += 1
    if text[i] == ' ':
        count_words += 1
    if text[i] == '.' or text[i] == '!' or text[i] == '?':
        count_Sentences += 1

# Calculate the level
L = count_letters * 100 / count_words
S = count_Sentences * 100 / count_words
index = 0.0588 * L - 0.296 * S - 15.8

# Print the result
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade ", round(index))

